src/bootstrap/builder.rs

```diff
@@ -616,32 +630,29 @@ impl<'a> Builder<'a> {
                 install::Rustc
             ),
             Kind::Run => describe!(run::ExpandYamlAnchors, run::BuildManifest, run::BumpStage0),
+            // These subcommands are special-cased in Build::build() and don't have typical steps
+            Kind::Format | Kind::Clean | Kind::Setup => vec![],
```

Comment:
I think this can be `unreachable!` instead, rather than silently doing nothing.
```suggestion
            Kind::Format | Kind::Clean | Kind::Setup => unreachable!("handled in Build::build()"),
```

Reply:
(this is not unreachable, related to the other problems below)